python/lsst/meas/algorithms/loadReferenceObjects.py

```diff
@@ -32,11 +34,513 @@
+        -------
+        md : `lsst.daf.base.PropertyList`
+        """
+        md = PropertyList()
+        md.add('RA', coord.getRa().asDegrees(), 'field center in degrees')
+        md.add('DEC', coord.getDec().asDegrees(), 'field center in degrees')
+        md.add('RADIUS', radius.asDegrees(), 'field radius in degrees, minimum')
+        md.add('SMATCHV', 1, 'SourceMatchVector version number')
+        filterName = "UNKNOWN" if filterName is None else str(filterName)
+        md.add('FILTER', filterName, 'filter name for photometric data')
+        md.add('EPOCH', "NONE" if epoch is None else epoch, 'Epoch (TAI MJD) for catalog')
+        return md
+
+    @staticmethod
+    def addFluxAliases(refCat, defaultFilter, filterReferenceMap):
+        """This function creates maps the input reference catalog into a new reference
+        catalog with added flux columns and aliases between camera and reference flux.
+
+        Parameters
+        ----------
+        refCat : `lsst.afw.table.SimpleCatalog`
+            Catalog of reference objects
+        defaultFilter : `str`
+            Name of the default reference filter
+        filterReferenceMap : `dict` of `str`
+            Dictionary with keys corresponding to a filter name, and values which
+            correspond to the name of the reference filter.
+
+        Returns
+        -------
+        refCat : `lsst.afw.table.SimpleCatalog`
+            Reference catalog with columns added to track reference filters
+
+        Raises
+        ------
+        `RuntimeError`
+            If specified reference filter name is not a filter specifed as a key in the
+            reference filter map.
+        """
+        refCat = ReferenceObjectLoader.remapReferenceCatalogSchema(refCat,
+                                                                   filterNameList=filterReferenceMap.keys())
+        aliasMap = refCat.schema.getAliasMap()
+
+        if filterReferenceMap is None:
+            filterReferenceMap = {}
+        for filterName, refFilterName in itertools.chain([(None, defaultFilter)],
+                                                         filterReferenceMap.items()):
+            if refFilterName:
+                camFluxName = filterName + "_camFlux" if filterName is not None else "camFlux"
+                refFluxName = refFilterName + "_flux"
+                if refFluxName not in refCat.schema:
+                    raise RuntimeError("Unknown reference filter %s" % (refFluxName,))
+                aliasMap.set(camFluxName, refFluxName)
+
+                refFluxErrName = refFluxName + "Sigma"
+                camFluxErrName = camFluxName + "Sigma"
+                aliasMap.set(camFluxErrName, refFluxErrName)
+
+        return refCat
+
+    @staticmethod
+    def remapReferenceCatalogSchema(refCat, *, filterNameList=None, position=False, photometric=False):
+        """This class takes in a reference catalog and maps it to a catalog with additional
+        columns.
```

Comment:
This is not a class.  Also probably better to use a different verb than "maps", which is a bit unclear.